var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -62,6 +74,49 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cmake@3.18:", type="build")
 
+    # Profiling
+    depends_on("hiop+deepchecking build_type=RelWithDebInfo", when="+hiop+debug")
```

Comment:
@CameronRutherford Why do you need to introduce an alias variant? Can't you do:
```suggestion
    depends_on("hiop+deepchecking build_type=RelWithDebInfo", when="+hiop build_type=RelWithDebInfo")
```

Reply:
@CameronRutherford if I understand correctly then this means the variants and hence the flags are not needed at all. So I just need to control things using the build type of exago and hiop. Long story short remove the variants and have the packages build types controlled by Exago build type.

Reply:
Yes. You've got it